integration_tests/src/main/java/com/nvidia/spark/rapids/tests/TimeoutSparkListener.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tests;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.scheduler.SparkListener;
+import org.apache.spark.scheduler.SparkListenerApplicationEnd;
+import org.apache.spark.scheduler.SparkListenerJobEnd;
+import org.apache.spark.scheduler.SparkListenerJobStart;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TimeoutSparkListener extends SparkListener {
+  private final Logger LOG = LoggerFactory.getLogger(TimeoutSparkListener.class);
+  private final JavaSparkContext sparkContext;
+  private final int timeoutSeconds;
+  private final ScheduledExecutorService runner = Executors.newScheduledThreadPool(1);
```

Comment:
Overall very nice! 
nit: Why not use the [newSingleThreadScheduledExecutor()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newSingleThreadScheduledExecutor--) method instead?

Reply:
@razajafri Why do you see the constraint for the number of threads (1) warranted? I do not see it necessary.